### ISSUE TITLE
CMake's `_USE_EXTERNAL` variables are confusing

### ISSUE THREAD

**Issue by mathstuf** [HDFGroup/hdf5#3637]
**Describe the bug**
The `ZLIB_USE_EXTERNAL` and `SZIP_USE_EXTERNAL` variables are confusingly named. They refer to "make a new external build and use it", not "use an external copy that has already been built". The hint in the docstring is that it uses "building", not "build", but this is far too subtle IMO. Especially when the behavior is such that if something fails, the associated functionality is just turned off with a warning message. If the functionality was requested, it should be a hard error if it cannot be made available to avoid surprises.

**Expected behavior**
If a feature is requested (say, `HDF5_ENABLE_Z_LIB_SUPPORT`), it should be an error if it is not found. Same with SZIP or any other filter requests. Additionally, docstrings and/or variable names could be more clear.

**Platform (please complete the following information)**
 - HDF5 version: 1.14.1
 - OS and version: macOS, Windows, Linux
 - Compiler and version: any
 - Build system (e.g. CMake, Autotools) and version: CMake
 - Any configure options you specified
 - MPI library and version (parallel HDF5): either; MPICH or MS-MPI

**Comment by derobins** [comment]
Agree this requires discussion. We're going to be setting up a recurring working group meeting to discuss filter issues (details TBD) and this should be one of the first things we discuss.

**Comment by byrnHDF** [comment]
Looking at this, These options control whether cmake will only do a find package or use the EXTERNALPROJECT_ADD (now FETCHCONTENT) process.
I think the name could be changed to *_USE_FINDPACKAGE_ONLY, which would be the opposite logic.

**Comment by mathstuf** [comment]
Yep, that's what I thought they were at first (not that it was related to `ExternalProject`). The docstring didn't clue me in either.

**Comment by byrnHDF** [comment]
Would adding, EXTERNALPROJECT_ADD (now FETCHCONTENT), help?

**Comment by byrnHDF** [comment]
OR change the name?

**Comment by mathstuf** [comment]
For me, yes, that in the docstring would have helped. Not sure how useful it would be for non-CMake experts building HDF5 though. Something like:

```
option(HDF5_BUILD_OWN_<PKG> "Build a copy of <PKG> for use during the build" OFF)
```

would be more mechanism-agnostic too.

**Comment by byrnHDF** [comment]
Should be clearer with fix in #4512,
